ivy_tests/test_ivy/test_functional/test_core/test_creation.py

```diff
@@ -202,11 +202,32 @@ def _asarray_helper(draw):
             shared_dtype=True,
         )
     )
+    nested_structure = draw(st.booleans())
+    if nested_structure:
+        sh = draw(
+            helpers.lists(
+                x=st.integers(min_value=1, max_value=20), min_size=1, max_size=5
+            )
+        )
+        sh = ivy.Shape(sh)
+        np_array = np.arange(3).astype(x_dtype[0])
+        ivy_array = ivy.array(x[0], dtype=x_dtype[0])
```

Comment:
we are generating the ivy array using the same function that we are testing, so I guess this might cause a loophole for failures to be missed

Reply:
In the meeting point @vedpatwardhan had mentioned ivy Array too as a leaf entry. Maybe he can comment on this.

Besides this @MahmoudAshraf97 please feel free to make such minor changes as you see fit. If there is an issue with overall nested structure creation logic then surely I will try to fix that. Thanks

Reply:
Actually, the points were prepared by @sherry30, but I agree with @Ishticode, unless there's something fundamentally wrong in creating a single leaf array, then we probably shouldn't consider it as a loophole. And adding to that, I don't think there's any other appropriate way to create an `ivy.Array` either (as we shouldn't be using the constructor). Thanks @MahmoudAshraf97 and @Ishticode 🙂

Reply:
This point is valid if we are trying to check if using an `ivy.Array` input will cause an exception, but what if the array is created correctly but with wrong contents? this mistake is going to occur in the ground truth example and in the test result and thus no errors will be caught because the result matches the ground truth, btw ivy arrays are converted to native arrays before going into the function so what applies to ivy arrays also apply to native arrays, that's why I see that we should instead create native arrays, at least we can guarantee that the native array creation functions are working well as a ground truth

Reply:
Yeah but the same issue with creating an array with wrong contents also exists with a native array creation function too right? 
In fact given that the native array creation functions aren't exactly equivalent in terms of the way they deal with dtype inferring and default devices and stuff, we are at a larger risk of these inconsistencies if we treat the native functions as the ground truth. 
Ofc given that we're generating `np.ndarray` in our data generation with `dtype_and_values`, this might end up being consistent in terms of the dtype, but we'd still need some equivalent of an `asarray` function as our helper which calls the appropriate native array creation function. So probably we're better off just using `ivy.asarray` and if there's an inconsistency in array creation due to a problem in our backend implementation, that'll propagate through the tests anyway.
Apologies if I've not understood your point properly, feel free to point it out if that's the case. Thanks @MahmoudAshraf97 🙂